File: MiracleMileAPI/MiracleMileAPI/Model/Like.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace MiracleMileAPI.Model
{
    public class Like
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public int ObjectId { get; set; }
        public TypeOfLike Type { get; set; }
        public bool Active { get; set; }
        public DateTime Created { get; set; }
    }
}

File: TopGun/src/app/model/like.ts
import { TypeOfLike } from "./type-of-like";

export class Like {
    public id: number;
    public UserId: number;
    public ObjectId: number;
    public Type : TypeOfLike;
    public Active : boolean;

    constructor(data: Partial<Like>) {
        Object.assign(this, data);
    }

    public static empty(): Like {
        return new Like({
            id: 0,
            UserId: 0,
            ObjectId: 0,
            Type: 0,
            Active: true,
        });
    }

}
